packages/components/src/components/post-popover/post-popover.tsx

```diff
@@ -1,22 +1,39 @@
 import { Component, Element, h, Host, Method, Prop } from '@stencil/core';
 import { Placement } from '@floating-ui/dom';
 import { version } from '../../../package.json';
+import { getAttributeObserver } from '../../utils/attribute-observer';
 
 /**
  * @slot default - Slot for placing content inside the popover.
  */
 
+let popoverInstances = 0;
+const popoverTargetAttribute = 'data-popover-target';
+
+const globalToggleHandler = (e: PointerEvent | KeyboardEvent) => {
+  const target = e.target as HTMLElement;
+  if (!target || !('getAttribute' in target)) return;
+  const popoverTarget = target.getAttribute(popoverTargetAttribute);
+  if (!popoverTarget || popoverTarget === '') return;
+  if ('key' in e && e.key !== 'Enter') return;
```

Comment:
It's not easy to read. Maybe you can extract those conditions into variables and have a single if condition?